engines/ags/engine/main/config.cpp

```diff
@@ -323,7 +325,22 @@ void apply_config(const ConfigTree &cfg) {
 		Common::String translation;
 		if (ConfMan.getActiveDomain()->tryGetVal("translation", translation) && !translation.empty())
 			_GP(usetup).translation = translation;
-		else
+		else if (!ConfMan.get("language").empty() && ConfMan.isKeyTemporary("language")) {
+			// Map the language defined in the command-line "language" option to its description
+			Common::Language lang = Common::parseLanguage(ConfMan.get("language"));
+			if (lang != Common::Language::UNK_LANG) {
+				translation = Common::getLanguageDescription(lang);
+			}
+
+			// Check if the game actually has such a translation, and set it if it does
+			Common::StringArray traFileNames = AGSMetaEngine::getGameTranslations(ConfMan.getActiveDomainName());
+			for (Common::StringArray::iterator iter = traFileNames.begin(); iter != traFileNames.end(); ++iter) {
+				if (translation.equalsIgnoreCase(*iter)) {
```

Comment:
The name of translation files can be anything, but in general they are one of:
- English name of the language, for example `French.tra` or `Spanish.tra`
- Translated name of the language, for example `polsky.tra`, `francais.tra`, or `français.tra`.
- The language code, for example `FR.tra` or `DE.tra`.
- And these can be combined with a prefix or suffix, for example `Nelly_Polish.tra`, `english2.tra`

Here you are handling the first case. Handling all the cases is not possible, but it should be easy to handle the third case as well by comparing with the language code in addition to the language description.
Also you could handle the fourth case by checking if `(*iter)` contains `translation` instead of using `equals` (only for the description, as the 2-letter language code might give too many false positive). Unfortunately I think there is no case-insensitive `contains` function, but this could be done by converting both string to lowercase before the check.

The second case could maybe be handled by using TranslationManager and adding the language descriptions to the strings to translate. But I think for now we can ignore it. If it works for 80% of the cases that will already be a good improvement.

Reply:
Thanks for the detailed explanation! Implemented cases 1, 3 and 4, did not implement 2, as noted